src/context/WalletProvider/TallyHo/components/Connect.tsx

```diff
@@ -71,13 +68,8 @@ export const TallyHoConnect = ({ history }: TallyHoSetupProps) => {
       try {
         const deviceId = await wallet.getDeviceID()
 
-        // Switch to Mainnet if wallet is on any other chain
-        const chainId = await wallet.ethGetChainId?.()
-        if (bnOrZero(chainId).toString() !== CHAIN_REFERENCE.EthereumMainnet) {
-          // TODO: Remove this comment when Tally multi-chain support is released
-          // This block is currently unreachable, Tally multi-chain support is currently under development
-          // Until this is supported in the published Tally extension, users will never be in a chain other than mainnet
-          await wallet.ethSwitchChain?.(bn(CHAIN_REFERENCE.EthereumMainnet).toNumber())
+        if (provider?.chainId !== '0x1') {
```

Comment:
Cherry-pick hunks of f304b4f416a99472525067a96240179fe1cc4855 to use `ethGetChainId()`. Don't error out since we can never be in a wrong chain to start with, add a similar TODO explaining this